quickstart.rb

```diff
@@ -45,26 +43,68 @@ def authorize
 apiCall = Google::Apis::DriveV3::DriveService.new
 apiCall.client_options.application_name = APPLICATION_NAME
 apiCall.authorization = authorize
-# List the 1000 most recently modified files.
+# List the most recently modified GIF files.
 response = apiCall.list_files(page_size: 10,
                               fields: 'nextPageToken, files(id, name, kind, mime_type)',
                               q: "mimeType='image/gif'",
                               spaces: 'photos'
                               )
 
 
-
+# Display output
 puts 'Files:'
 puts 'No files found' if response.files.empty?
+
 response.files.each do |file|
-  #puts file.inspect
-  #puts
-  puts "#{file.name} (#{file.id}) (#{file.kind}) ((#{file.mime_type}))"
 
-  puts "================="
+  #puts file.inspect
+  #puts "#{file.name} (#{file.id}) (#{file.mime_type})"
+  puts ""
   aFile = apiCall.get_file(file.id,
-                            {fields: 'id, name, kind, mime_type, thumbnailLink, webContentLink, webViewLink'}
+                            {fields: 'id, name, mimeType, thumbnailLink, webViewLink, webContentLink, createdTime, ownedByMe, size, imageMediaMetadata'}
                           )
-  puts aFile.inspect
-  puts "==========="
+
+
+
+  puts "File: #{aFile.name}"
+  puts "ID: #{aFile.id}"
+  puts "Created: #{aFile.created_time}"
+  puts "View: #{aFile.web_view_link}"
+  puts "Download: #{aFile.web_content_link}"
+  puts "Thumbnail #{aFile.thumbnail_link}"
+  puts "Mine: #{aFile.owned_by_me}"
+  puts "Dimensions: #{aFile.image_media_metadata.width}x#{aFile.image_media_metadata.height}"
+  puts "Size: #{aFile.size}bytes"
+  if aFile.image_media_metadata.location != nil
+    puts "Loc: #{aFile.image_media_metadata.location.latitude}, #{aFile.image_media_metadata.location.longitude}"
+  else
+    puts "Loc: 0, 0"
+  end
+
+  #puts aFile.inspect
 end
+
+# note: I don't think exporting of gifs works. :(
```

Comment:
The export method is to download the file on a different format, Ex: from .doc to PDF.

[Docs here](https://developers.google.com/drive/v3/web/about-sdk#export_and_convert_google_docs)

Reply:
yup. boo! 👎